src/devices/Hcsr04/Hcsr04.cs

```diff
@@ -147,10 +150,13 @@ private bool TryGetDistance(out double result)
         /// <inheritdoc/>
         public void Dispose()
         {
-            if (_controller != null)
+            if (_shouldDispose)
             {
```

Comment:
NIT: you can also change the bellow code to just be:

```c#
_controller?.Dispose();
_controller = null;
```
in order to remove the if-check